third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/WGMMA.cpp

```diff
@@ -271,27 +271,27 @@ llvm::SmallVector<Value> loadReg(ConversionPatternRewriter &rewriter,
   OpBuilder::InsertionGuard g(rewriter);
   rewriter.setInsertionPoint(insertBefore);
 
-  // Internally bfloat16 is stored as int16, so we check them both
-  auto isBF16 =
-      elements[0].getType().isBF16() || elements[0].getType().isInteger(16);
-
-  if (!(elements[0].getType().isF16() || isBF16)) {
+  bool needPacking = elements[0].getType().isIntOrFloat() &&
+                     elements[0].getType().getIntOrFloatBitWidth() < 32;
+  if (!needPacking) {
     llvm::SmallVector<Value> mmaOut(numElements);
     for (int i = 0; i < numElements; ++i)
       mmaOut[i] = elements[startIndex + i];
     return mmaOut;
   }
+  Type elementType = elements[0].getType();
+  int numElementsPacked = 32 / elementType.getIntOrFloatBitWidth();
```

Comment:
Should be numPackedElements, but actually that doesn't match what this is.  elemsPer32Bits or numElemsPer32Bits.  Or you could s/32Bits/word/ and define the word size in a comment.  Then below it's simply numWords.

Reply:
well it is the number of elements we pack together

Reply:
To me "num elements packed" sounds like the same thing as num32BitValues, i.e. the number of elements after packing, but up to you.

Reply:
ah I see what you mean, sure changing the name